tsinfer/inference.py

```diff
@@ -1912,6 +1908,146 @@ def get_mutations(self, node_id):
         return self.mutations[node_id]
 
 
+def has_single_edge_over_grand_root(ts):
+    # Internal function to check if this is a "raw" inferred tree sequence.
+    if ts.num_edges < 2:
+        # must have edge to grand root and above grand root
+        return False
+    last_edge = ts.edge(-1)
+    if last_edge.left != 0 or last_edge.right != ts.sequence_length:
+        return False  # Not a single edge spanning the entire genome
+    if ts.edge(-2).parent == last_edge.parent:
+        return False  # other edges point to the oldest node => not a virtual-like root
+    return True
+
+
+def has_same_root_everywhere(ts):
+    roots = set()
+    for tree in ts.trees():
+        if not tree.has_single_root:
+            return False
+        roots.add(tree.root)
+        if len(roots) > 1:
+            return False
+    return True
+
+
+def post_process(
+    ts,
+    *,
+    split_mrca=None,
+):
+    """
+    Post-process a tsinferred tree sequence into a more conventional form. This is
+    the function run by default on the final tree sequence output by
+    :func:`match_samples`. It involves removing the oldest node if it sits above a
+    grand MRCA ancestor (this "virtual-root-like node" is used to enable LS matching)
+    then splitting up the grand MRCA (if split_mrca is ``True``), and finally
+    reordering the sample nodes such that they are the first nodes listed in the
+    node table,  removing tree nodes and edges that are not on a path between the
+    root and any of the samples (by applying the :meth:`tskit.TreeSequence.simplify`
+    method with ``keep_unary`` set to True but ``filter_sites``, ``filter_populations``
+    and ``filter_individuals`` set to False).
+
+    :param bool split_mrca: If the oldest node is only a parent to a single
```

Comment:
Move the bulk of this discussion to a paragraph rather than within the ``split_mrca`` parameter description - these should be short and focus on things like types, etc.